front/src/utils/checkSingleStatus.js

```diff
@@ -0,0 +1,33 @@
+export const checkSingleStatus = (selectSettingsStatus) => {
```

Comment:
W sumie nie potrzbujesz w ogóle ten funkcji, wystarczy sam selektor:
export const selectBooksFetchStatus = (state) => state.books.dataStatus

Trochę jest bez sensu, bo przekazujesz status selectSettingsStatus, potem robisz z niego tablicę jednoelementową w linii 2, po której iterujesz forEachem, zastepujac ten status. Pokomplikowane niepotrzebnie, finalnie masz te same informacje co w dataStatus.

W komponencie Grid/index.js zrób 
const status = useSelector(selectBooksFetchStatus);

a potem:

if (status === "failed") {
    return "Fetching data error...";
  }

  if (status === "inProgress") {
    return ""; --- tu by sie przydał jakiś loader, np. CircularProgress z materiala
  }

  if (!data.isLoaded) { --- to jest niepotrzebne, operuj na samym data (if(!data))
    return null;
  }

Reply:
Done.